src/Query.re

```diff
@@ -93,5 +100,12 @@ let softCompoundDelete = (baseQuery, table, decoder, id, conn) => {
     WHERE $table.`id` = ?
   |j};
   Sql.Promise.mutate(conn, ~sql, ~params?, ())
-  |> Js.Promise.then_((_) => getById(baseQuery, table, decoder, id, conn));
+  |> Js.Promise.then_(((success, _)) =>
```

Comment:
What if more than one row was affected?  `NotFound` isn't the correct response.  You should use the `bs-result` libraries `result` type so that we can standardize across libraries.